### PR TITLE
Limit order TEAL template

### PR DESCRIPTION
**Pull request by EvanJRichard**
## Summary
This template implements `limit-order-a.teal.impl` from `go-algorand`.

### Testing
Automatic unit/sdk testing. Added a golden test for program and address.

### See also
https://github.com/algorand/go-algorand-sdk/issues/98

### CODE REVIEW

File: templates/limitOrder.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+	var signedGroup []byte
+	signedGroup = append(signedGroup, assetsForAlgosSigned...)
+	signedGroup = append(signedGroup, algosForAssetsSigned...)
+
+	return signedGroup, nil
+}
+
+// MakeLimitOrder allows a user to exchange some number of N assets for D algos.
+// This is a contract account.
```
Comment (rotemh) [inline templates/limitOrder.go]: This is a bit of a weird comment here that makes much more sense in the TEAL code.

File: templates/limitOrder.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	}
+
+	var signedGroup []byte
+	signedGroup = append(signedGroup, assetsForAlgosSigned...)
+	signedGroup = append(signedGroup, algosForAssetsSigned...)
+
+	return signedGroup, nil
+}
+
+// MakeLimitOrder allows a user to exchange some number of N assets for D algos.
```
Comment (rotemh) [inline templates/limitOrder.go]: This line and line 75 are contradicting

File: templates/limitOrder.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+// secretKey: secret key for signing transactions
+// fee: fee per byte used for the transactions
+// algoAmount: number of algos to transfer
+// firstRound: first round on which these txns will be valid
+// lastRound: last round on which these txns will be valid
+// genesisHash: genesisHash indicating the network for the txns
+// the first payment sends money (Algos) from contract to the recipient (we'll call him Buyer), closing the rest of the account to Owner
+// the second payment sends money (the asset) from Buyer to the Owner
+// these transactions will be rejected if they do not meet the restrictions set by the contract
+func (lo LimitOrder) GetSwapAssetsTransaction(assetAmount uint64, contract, secretKey []byte, fee, algoAmount, firstRound, lastRound uint64, genesisHash []byte) ([]byte, error) {
```
Comment (rotemh) [inline templates/limitOrder.go]: This one does the correct thing!

Comment (winder) [reply templates/limitOrder.go]: we should probably rename `algoAmount` to `microAlgoAmount`

**Pull request merged**